service/controller/resource/heartbeatrouting/receiver/receiver_test.go

```diff
@@ -88,6 +88,40 @@ func TestEnsureReceiver(t *testing.T) {
 			len:            1,
 			index:          0,
 		},
+		{
+			// The HTTPConfig from global config should be merged when creating HTTPConfig for the receiver
+			name: "update merges global proxy settings",
+			cfg: alertmanagerconfig.Config{
+				Global: &alertmanagerconfig.GlobalConfig{
+					HTTPConfig: &promcommonconfig.HTTPClientConfig{
+						ProxyURL: promcommonconfig.URL{URL: &url.URL{Host: "proxyhost:8080"}},
+					},
+				},
+				Receivers: []*alertmanagerconfig.Receiver{
+					&alertmanagerconfig.Receiver{
+						Name: "heartbeat_installation_cluster",
+						WebhookConfigs: []*alertmanagerconfig.WebhookConfig{
+							&alertmanagerconfig.WebhookConfig{
+								URL: &alertmanagerconfig.URL{
+									URL: u,
+								},
+								HTTPConfig: &promcommonconfig.HTTPClientConfig{
+									BasicAuth: &promcommonconfig.BasicAuth{
+										Password: "secret-key",
+									},
+								},
```

Comment:
Where is the result of this test with the merged proxy url ?

Reply:
If I'm reading this correctly, the way these test are structured is we don't specify the expected result, we just specify the source and that we expect something to change.

I'd be open to refactoring this a bit to be a bit more explicit and specify the expected result, WDYT?

Reply:
This test ensure, that the whole update logic is correct and does update only when it needs to.
Maybe write a different test for this, because here we want to test that the proxy url is propagated from global to receiveirs, its a different test IMO.

Reply:
Done.